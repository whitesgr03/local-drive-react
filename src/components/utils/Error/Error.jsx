// Packages
import { Link, useLocation } from 'react-router-dom';
import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

// Styles
import styles from './Error.module.css';
import { icon } from '../../../styles/icon.module.css';

export const Error = ({ error, onError, children }) => {
	const [darkTheme, setDarkTheme] = useState(false);
	const { state } = useLocation();

	const message = state?.error || error;

	message && console.error('Error component:', message);

	const handleClick = () => {
		document.body.removeAttribute('style');
	};

	useEffect(() => {
		const getColorTheme = () => {
			const darkScheme = localStorage.getItem('darkTheme');

			const browserDarkScheme =
				window.matchMedia('(prefers-color-scheme: dark)')?.matches ?? false;

			darkScheme === null &&
				localStorage.setItem('darkTheme', browserDarkScheme);

			setDarkTheme(
				darkScheme === null ? browserDarkScheme : darkScheme === 'true',
			);
		};
		!children && getColorTheme();
	}, [children]);

	return (
		<div className={`${styles.container} ${darkTheme ? 'dark' : ''}`}>
			<div className={styles.error}>
				<span className={`${icon} ${styles.alert}`} />
				<div className={styles.message}>
					<p>Our apologies, there has been an error.</p>
					{children ? (
						children
					) : (
						<p>
							Please come back later, or if you have any questions, contact us.
						</p>
					)}
				</div>
				{onError ? (
					<button className={styles.link} onClick={() => onError(null)}>
						Go back
					</button>
				) : (
					<Link to="/" className={styles.link} onClick={handleClick}>
						Back to Home Page
					</Link>
				)}
			</div>
		</div>
	);
};

Error.propTypes = {
	error: PropTypes.string,
	children: PropTypes.node,
	onError: PropTypes.func,
};
