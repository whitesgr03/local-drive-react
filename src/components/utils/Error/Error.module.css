.container {
	display: flex;
	width: 100%;
	height: 100%;
}

.error {
	display: flex;
	row-gap: 30px;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	background-color: var(--light-scheme-main-bgc);
	padding: 5%;
	width: 100%;
	color: var(--light-scheme-error);
	text-align: center;
}

.error p:first-of-type {
	display: flex;
	column-gap: 10px;
	align-items: center;
}

.alert {
	mask-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 36 36'%3E%3Cpath fill='%23000' d='M18 2.5c-8.82 0-16 6.28-16 14s7.18 14 16 14a18 18 0 0 0 4.88-.68l5.53 3.52a1 1 0 0 0 1.54-.84v-6.73a13 13 0 0 0 4-9.27C34 8.78 26.82 2.5 18 2.5m10.29 22.11a1 1 0 0 0-.32.73v5.34l-4.38-2.79a1 1 0 0 0-.83-.11a16 16 0 0 1-4.76.72c-7.72 0-14-5.38-14-12s6.28-12 14-12s14 5.38 14 12a11.08 11.08 0 0 1-3.71 8.11' class='clr-i-outline clr-i-outline-path-1'/%3E%3Cpath fill='%23000' d='M18 20.63a1 1 0 0 0 1-1V8.48a1 1 0 1 0-2 0v11.13a1 1 0 0 0 1 1.02' class='clr-i-outline clr-i-outline-path-2'/%3E%3Ccircle cx='18' cy='24.04' r='1.33' fill='%23000' class='clr-i-outline clr-i-outline-path-3'/%3E%3Cpath fill='none' d='M0 0h36v36H0z'/%3E%3C/svg%3E");
	width: var(--icon-size-desktop);
	height: var(--icon-size-desktop);
}

.message {
	display: flex;
	row-gap: 10px;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.link {
	color: var(--light-scheme-color);
	text-decoration: underline;
}

/* Dark scheme classes */

:global(.dark) .error {
	background-color: var(--dark-scheme-layout-bgc);
	color: var(--dark-scheme-error);
}

:global(.dark) .link {
	color: var(--dark-scheme-color);
}
