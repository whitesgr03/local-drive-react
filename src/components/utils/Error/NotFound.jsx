import { Link } from 'react-router-dom';

// Styles
import style from './Error.module.css';
import { icon } from '../../../styles/icon.module.css';

export const NotFound = () => {
	return (
		<div className={style.error}>
			<span className={`${icon} ${style.alert}`} />
			<h2>404 Page Not Found</h2>
			<div className={style.message}>
				<p>The page you are looking for could not be found.</p>
				<p>You may surf over to our other pages.</p>
			</div>
			<Link to="/" className={style.link}>
				Back to Home Page
			</Link>
		</div>
	);
};
