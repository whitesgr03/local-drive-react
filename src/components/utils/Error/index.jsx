// Packages
import { Link, useLocation } from 'react-router-dom';
import PropTypes from 'prop-types';

// Styles
import style from './Error.module.css';
import { icon } from '../../../styles/icon.module.css';

export const Error = ({ error }) => {
	const { state } = useLocation();
	console.error('Error component:', state ? state.error : error);

	return (
		<div className={style.error}>
			<span className={`${icon} ${style.alert}`} />
			<div className={style.message}>
				<p>Our apologies, there has been an error.</p>
				<p>Please come back later, or if you have any questions, contact us.</p>
			</div>
			<Link to="/" className={style.link}>
				Back to Home Page
			</Link>
		</div>
	);
};

Error.propTypes = {
	error: PropTypes.string,
};
