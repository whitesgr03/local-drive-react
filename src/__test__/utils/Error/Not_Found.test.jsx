import { expect, describe, it } from 'vitest';
import { render } from '@testing-library/react';

import { RouterProvider, createMemoryRouter } from 'react-router-dom';

import { Not_Found } from '../../../components/utils/Error/Not_Found';

describe('Not_Found component', () => {
	it('should match snapshot', () => {
		const router = createMemoryRouter([
			{
				path: '/',
				element: <Not_Found />,
			},
		]);

		const { asFragment } = render(<RouterProvider router={router} />);

		const actual = asFragment();

		expect(actual).toMatchSnapshot();
	});
});
