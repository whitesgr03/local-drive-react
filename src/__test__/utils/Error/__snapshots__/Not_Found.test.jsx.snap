// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_error_48ab16"
  >
    <span
      class="undefined _alert_48ab16"
    />
    <h2>
      404 Page Not Found
    </h2>
    <div
      class="_message_48ab16"
    >
      <p>
        The page you are looking for could not be found.
      </p>
      <p>
        You may surf over to our other pages.
      </p>
    </div>
    <a
      class="_link_48ab16"
      href="/"
    >
      Back to Home Page
    </a>
  </div>
</DocumentFragment>
`;
